### PR TITLE
Fixes #29439 - VMRC doesn't start

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Review by timogoebel — DISMISSED** [review]
Not pretty, but should fix the issue.

@ezr-ondrej: WDYT?

**Comment by tbrisker** [comment]
@sbernhard can you explain how setting the timeout fixes this issue?

**Comment by sbernhard** [comment]
the browser behavior is:
- execute the onclick
- execute the href

if the onclick sets disabled=disabled the href won't be executed anymore. the setTimeout() does this asynchronously and returns immediately "true" so that the "href" is executed.

**Comment by tbrisker** [comment]
Why do we need to disable it at all then? iiuc it redirects to the VMRC url - so the page would reload anyways and the button won't be visible

**Comment by sbernhard** [comment]
disable it, so that the users is only able to press the button once. 
vmrc link will start a external application in a separate window.

**Review by chris1984 — DISMISSED** [review]

**Comment by sbernhard** [comment]
ping!

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
@tbrisker can this get a merge please? Two acks and all tests are green.

**Comment by timogoebel** [comment]
@chris1984: Let's wait for a final ACK from @ezr-ondrej.

**Comment by ezr-ondrej** [comment]
I'm sorry it took me a while. It works, but I don't like it. Merging and I'll try to figure something later.

File: app/views/hosts/console/vmrc.html.erb
Comment (ezr-ondrej) [inline app/views/hosts/console/vmrc.html.erb]:
```suggestion
    <%= link_to(_('Launch Console'),
                @console[:console_url],
                :class => 'btn btn-primary btn-lg',
                :data => {
                  :disable_with => "<i class='fa fa-spinner fa-spin'></i> #{_('Opening VMRC Console')}..."
                }) %>
```
It would result in:
image

Drawback is, that it's not easy to set a disabled style on btns disabled by `jquery-ujs`, it looks ok to me though. @sbernhard WDYT?

Comment (sbernhard) [reply app/views/hosts/console/vmrc.html.erb]: if it works (in firefox and chrome), I'm happy :)

Comment (ezr-ondrej) [reply app/views/hosts/console/vmrc.html.erb]: Yeah, it works OK in both. Will you update your PR?

**Comment by ezr-ondrej** [comment]
@timogoebel could you just ack if you're ok with this? As I proposed the code.

**Review by timogoebel — DISMISSED** [review]
I love it, thanks!

**Comment by ezr-ondrej** [comment]
I'll rebase and squash for you @sbernhard, redmine issue checker don't like the suggestion application.

**Comment by ezr-ondrej** [comment]
Ok, I can't push there :( could you squash @sbernhard ? Without the redmine check I can't merge.

**Review by timogoebel — APPROVED** [review]

**Review by ezr-ondrej — APPROVED** [review]
Thanks @sbernhard, @timogoebel and @chris1984 !
And sorry for blocking this for a while.

**Comment by ezr-ondrej** [comment]
cherry-pick for:
- 2.0: #7579
- 1.24: #7580

**Pull request merged**
